src/azure-cli/azure/cli/command_modules/storage/operations/account.py

```diff
@@ -337,8 +337,16 @@ def update_management_policies(client, resource_group_name, account_name, parame
     return client.create_or_update(resource_group_name, account_name, policy=parameters)
 
 
-# TODO: support updating other properties besides 'enable_change_feed'
-def update_blob_service_properties(cmd, instance, enable_change_feed=None):
+# TODO: support updating other properties besides 'enable_change_feed,delete_retention_policy'
+def update_blob_service_properties(cmd, instance, enable_change_feed=None, delete_retention=None,
+                                   delete_retention_period=None):
     if enable_change_feed is not None:
         instance.change_feed = cmd.get_models('ChangeFeed')(enabled=enable_change_feed)
+    if delete_retention is not None or delete_retention_period is not None:
+        if instance.delete_retention_policy is None:
+            instance.delete_retention_policy = cmd.get_models('DeleteRetentionPolicy')
+        if delete_retention is not None:
+            instance.delete_retention_policy.enabled = delete_retention
+        if delete_retention_period is not None:
+            instance.delete_retention_policy.days = delete_retention_period
```

Comment:
According to your current logic, users can set delete retention days when no delete retention service enabled. It may be not reasonable. Please re-think about it.

Reply:
They can't set '--delete-retention-days' when delete retention service disabled in PowerShell, so I changed this logic to be consistent with PowerShell